scipy/linalg/tests/test_decomp.py

```diff
@@ -2725,57 +2725,79 @@ def test_subspace_angles():
 
 class TestCDF2RDF(object):
 
+    def matmul(self, a, b):
+        return np.einsum('...ij,...jk->...ik', a, b)
+
+    def assert_eig_valid(self, w, v, x):
+        assert_array_almost_equal(
+            self.matmul(v, w),
+            self.matmul(x, v)
+        )
+
+    def test_single_array0x0real(self):
+        # eig doesn't support 0x0 in old versions of numpy
+        X = np.empty((0, 0))
+        w, v = np.empty(0), np.empty((0, 0))
+        wr, vr = cdf2rdf(w, v)
+        self.assert_eig_valid(wr, vr, X)
+
     def test_single_array2x2_real(self):
         X = np.array([[1, 2], [3, -1]])
         w, v = np.linalg.eig(X)
         wr, vr = cdf2rdf(w, v)
-        assert_array_almost_equal(vr.dot(wr), X.dot(vr))
+        self.assert_eig_valid(wr, vr, X)
 
     def test_single_array2x2_complex(self):
         X = np.array([[1, 2], [-2, 1]])
         w, v = np.linalg.eig(X)
         wr, vr = cdf2rdf(w, v)
-        assert_array_almost_equal(vr.dot(wr), X.dot(vr))
+        self.assert_eig_valid(wr, vr, X)
 
     def test_single_array3x3_real(self):
         X = np.array([[1, 2, 3], [1, 2, 3], [2, 5, 6]])
         w, v = np.linalg.eig(X)
         wr, vr = cdf2rdf(w, v)
-        assert_array_almost_equal(vr.dot(wr), X.dot(vr))
+        self.assert_eig_valid(wr, vr, X)
 
     def test_single_array3x3_complex(self):
         X = np.array([[1, 2, 3], [0, 4, 5], [0, -5, 4]])
         w, v = np.linalg.eig(X)
         wr, vr = cdf2rdf(w, v)
-        assert_array_almost_equal(vr.dot(wr), X.dot(vr))
+        self.assert_eig_valid(wr, vr, X)
+
+    def test_random_1d_stacked_arrays(self):
+        # cannot test M == 0 due to bug in old numpy
+        for M in range(1, 7):
+            X = np.random.rand(10000, M, M)
+            w, v = np.linalg.eig(X)
+            wr, vr = cdf2rdf(w, v)
+            self.assert_eig_valid(wr, vr, X)
 
-    def test_random_stacked_arrays(self):
-        N = int(1e4)
-        for M in range(2, 7):
-            X = np.random.rand(N, M, M)
+    def test_random_2d_stacked_arrays(self):
+        # cannot test M == 0 due to bug in old numpy
+        for M in range(1, 7):
+            X = np.random.rand(100, 100, M, M)
```

Comment:
Smaller test matrices would be better --- no need to make the test runtime close to a second.

Reply:
This test was already testing 10000, but I suppose I've added a second one the same length. I'll shrink them to 100 and 10x10?